src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java

```diff
@@ -32,6 +33,7 @@
  * Checks that all packages have a package documentation. See the documentation
  * for more information.
  */
+@GlobalStatefulCheck
```

Comment:
For GlobalStateful it should be fine, but I don't remember how we are going to handle `beginProcessing` for other types of checks.
Each thread will have to create a new instance, so we will have to recall `beginProcessing` each instance?
My own implementation made a true, deep clone before passing it to the thread, so it was never an issue.

Reply:
`beginProcessing` and `finishProcessing` should be executed on each instance of module(in each thread) but before actual multi-thread processing is started.
Lets think about it in other PR/Issues. By this PR I just want to place annotations to other Checks and it is completely ok  to change annotations when we come to actual MT implementation.

So this PR is just declaration of intend, no obligations.

Reply:
Yes, I am just thinking out loud.